File: ios/MRT/DesignSystem/Components/GHButton.swift
import SwiftUI

struct GHButton: View {
    enum Style {
        case primary
        case secondary
        case danger
    }

    let title: String
    let icon: String?
    let style: Style
    let action: () -> Void

    var body: some View {
        Button(action: action) {
            HStack(spacing: GHSpacing.sm) {
                if let icon {
                    Image(systemName: icon)
                        .font(.system(size: 13, weight: .semibold))
                }

                Text(title)
                    .font(GHTypography.bodySm)
                    .fontWeight(.semibold)
            }
            .padding(.horizontal, GHSpacing.md)
            .padding(.vertical, GHSpacing.sm)
            .frame(minHeight: 36)
            .foregroundStyle(foregroundColor)
            .background(backgroundColor)
            .overlay(
                RoundedRectangle(cornerRadius: GHRadius.sm)
                    .stroke(borderColor, lineWidth: style == .secondary ? 1 : 0)
            )
            .clipShape(RoundedRectangle(cornerRadius: GHRadius.sm))
        }
        .buttonStyle(.plain)
    }

    private var backgroundColor: Color {
        switch style {
        case .primary:
            return GHColors.accentBlue.opacity(0.15)
        case .secondary:
            return GHColors.bgSecondary
        case .danger:
            return GHColors.accentRed.opacity(0.15)
        }
    }

    private var foregroundColor: Color {
        switch style {
        case .primary:
            return GHColors.accentBlue
        case .secondary:
            return GHColors.textSecondary
        case .danger:
            return GHColors.accentRed
        }
    }

    private var borderColor: Color {
        style == .secondary ? GHColors.borderDefault : .clear
    }
}

#Preview {
    ZStack {
        GHColors.bgPrimary.ignoresSafeArea()
        HStack {
            GHButton(title: "Primary", icon: "paperplane.fill", style: .primary) {}
            GHButton(title: "Secondary", icon: nil, style: .secondary) {}
        }
        .padding()
    }
}

File: android/app/src/main/java/com/mrt/app/designsystem/components/GHButton.kt
package com.mrt.app.designsystem.components

import androidx.compose.foundation.BorderStroke
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.defaultMinSize
import androidx.compose.foundation.layout.padding
import androidx.compose.material3.Button
import androidx.compose.material3.ButtonDefaults
import androidx.compose.material3.Icon
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.graphics.vector.ImageVector
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import com.mrt.app.designsystem.theme.GHColors
import com.mrt.app.designsystem.theme.GHType

enum class GHButtonStyle {
    Primary,
    Secondary,
    Danger,
}

@Composable
fun GHButton(
    text: String,
    onClick: () -> Unit,
    modifier: Modifier = Modifier,
    style: GHButtonStyle = GHButtonStyle.Primary,
    icon: ImageVector? = null,
    enabled: Boolean = true,
) {
    val backgroundColor = when (style) {
        GHButtonStyle.Primary -> GHColors.AccentBlue.copy(alpha = 0.15f)
        GHButtonStyle.Secondary -> GHColors.BgSecondary
        GHButtonStyle.Danger -> GHColors.AccentRed.copy(alpha = 0.15f)
    }
    val foregroundColor = when (style) {
        GHButtonStyle.Primary -> GHColors.AccentBlue
        GHButtonStyle.Secondary -> GHColors.TextSecondary
        GHButtonStyle.Danger -> GHColors.AccentRed
    }
    val border = if (style == GHButtonStyle.Secondary) {
        BorderStroke(1.dp, GHColors.BorderDefault)
    } else {
        null
    }

    Button(
        onClick = onClick,
        modifier = modifier.defaultMinSize(minHeight = 36.dp),
        enabled = enabled,
        shape = androidx.compose.foundation.shape.RoundedCornerShape(6.dp),
        border = border,
        colors = ButtonDefaults.buttonColors(
            containerColor = backgroundColor,
            contentColor = foregroundColor,
            disabledContainerColor = backgroundColor.copy(alpha = 0.4f),
            disabledContentColor = foregroundColor.copy(alpha = 0.5f),
        ),
    ) {
        Row(
            horizontalArrangement = Arrangement.spacedBy(8.dp),
            verticalAlignment = androidx.compose.ui.Alignment.CenterVertically,
        ) {
            if (icon != null) {
                Icon(imageVector = icon, contentDescription = null)
            }
            Text(
                text = text,
                style = GHType.BodySm,
                fontWeight = FontWeight.SemiBold,
                modifier = Modifier.padding(vertical = 2.dp),
            )
        }
    }
}
